### PR TITLE
Azure App Service - windows container deployment support

### PR DESCRIPTION
**Pull request by vincentdass**
image

- [ ] L0 test failure fix

### CODE REVIEW

File: Tasks/Common/AzureRmDeploy-common/operations/ContainerBasedDeploymentUtility.ts [hunk 1]
```diff
@@ -44,9 +44,14 @@ export class ContainerBasedDeploymentUtility {
             'value': startupCommand
         }
 
+        appSettingsNewProperties.windowsFxVersion = {
```
Comment (vincentdass) [inline Tasks/Common/AzureRmDeploy-common/operations/ContainerBasedDeploymentUtility.ts]: This value will be ignored by ARM call for Linux app and linuxFxVersion will be ignored for Windows app.

**Comment by SumiranAgg** [comment]
Why have we upgraded patch for function app and web app tasks? As per my understanding only container tasks will be affected.

**Comment by SumiranAgg** [comment]
Datasource has to be fixed to list both linux and windows type apps on container.

**Comment by SumiranAgg** [comment]
Will we be adding this support in old task?

**Comment by vincentdass** [comment]
> Why have we upgraded patch for function app and web app tasks? As per my understanding only container tasks will be affected.

there is a new CI check that expects task version for changed common modules are updated.

**Comment by vincentdass** [comment]
> Will we be adding this support in old task?

no :)

**Comment by vincentdass** [comment]
> Datasource has to be fixed to list both linux and windows type apps on container.

sure. working on that.

File: Tasks/Common/AzureRmDeploy-common/operations/ContainerBasedDeploymentUtility.ts (hunk 1 above)
Comment (vincentdass) [reply Tasks/Common/AzureRmDeploy-common/operations/ContainerBasedDeploymentUtility.ts]: changed logic

File: Tasks/AzureFunctionAppContainerV1/azurefunctiononcontainerprovider.ts [hunk 2]
```diff
@@ -29,9 +30,16 @@ export class AzureFunctionOnContainerDeploymentProvider{
         this.azureEndpoint = await new AzureRMEndpoint(this.taskParams.connectedServiceName).getEndpoint();
         console.log(tl.loc('GotconnectiondetailsforazureRMWebApp0', this.taskParams.WebAppName));
         
-        if(!this.taskParams.ResourceGroupName){
-            var appDetails = await AzureResourceFilterUtility.getAppDetails(this.azureEndpoint, this.taskParams.WebAppName);
+        if(!this.taskParams.ResourceGroupName) {
```
Comment (SumiranAgg) [inline Tasks/AzureFunctionAppContainerV1/azurefunctiononcontainerprovider.ts]: Should we move this code to taskparameters? For multi-container, we might do some checks on apptype before deployment provider is invoked.

Comment (vincentdass) [reply Tasks/AzureFunctionAppContainerV1/azurefunctiononcontainerprovider.ts]: removed that logic, as function app container has only linux

**Pull request merged**